src/meta_ops.rs

```diff
@@ -735,7 +736,113 @@ pub fn less_equal<'gc>(
     lhs: Value<'gc>,
     rhs: Value<'gc>,
 ) -> Result<MetaResult<'gc, 2>, MetaOperatorError> {
-    meta_metaop(ctx, lhs, rhs, MetaMethod::Le, |a, b| {
+    meta_metaop(ctx, lhs, rhs, MetaMethod::Le, |_, a, b| {
         Some(a.to_constant()?.less_equal(&b.to_constant()?)?.into())
     })
 }
+
+#[derive(Debug, Clone, Collect)]
+#[collect(no_drop)]
+pub struct ConcatMetaCall<'gc> {
+    pub function: Function<'gc>,
+    // Needs to be owned to work around borrowing issues in
+    // LuaFrame::call_meta_function
+    pub args: Vec<Value<'gc>>,
```

Comment:
I have a plan for something else to do here. The borrowing issue is real because it would reference a slice in a Lua frame stack and call a function that is expected to manipulate the stack.

Instead, we just shouldn't use `LuaFrame::call_meta_function` and make one specific for concat. Instead of return arguments here, or even having a `ConcatMetaCall` struct, `ConcatMetaResult` should just return a function, and the documentation should say that the function should be called on the `values` provided to `concat`.

Reply:
I've changed it to do this in the latest version, though you should check over the method (`call_meta_function_in_place`, name subject to change) to make sure that it lines up with what you expect.